### ISSUE TITLE
Bug with \Mollie_Mpm_Model_Observer::restoreQuoteWhenReturningFromMollie

### ISSUE THREAD

**Issue by Ig0r-M** [mollie/Magento#164]
**Given** following event
https://github.com/mollie/Magento/blob/18a6b259b186404ae0158caa65e8e46b843853a0/app/code/community/Mollie/Mpm/Model/Observer.php#L133-L150
**Then** if a user places an order
**And** presses back
**As** a result above code will make set active to true for quote that was used to place the order
**And** if we place another order
**Then** this will result in creating another invoice for the same order with value of 0
**Thus** creating a BUG that an order was not created

**Plus** observed is not checking if used method is actually Mollie and does this for all quotes

**Replicated** in 5.5.3

**Comment by Frank-Magmodules** [comment]
Hi @Ig0r-M , Thank you for opening this issue. We can't directly reproduce this. Can you contact us through our [contact form](https://www.magmodules.eu/support-form.html) so we can exchange log files to check the issue?
Thank you!

**Comment by Ig0r-M** [comment]
@Frank-Magmodules could you go over below steps, after this if you still cannot reproduce i will try to make a .gif indicating each step and values in db for quote.

**Steps to Reproduce**

- [ ] Enable check/money order `Mage_Payment_Block_Form_Checkmo`
- [ ] Have Mollie Mpm enabled
- [ ] Add product to the basket
  - [ ] go to checkout (quote is created and is active)
- [ ] Place an Order - previously created quote has `is_active` set to false
  - [ ] You will be redirected to Checkout Success page 
  - [ ] click back button form the browser on browser
![image](https://user-images.githubusercontent.com/22006850/75966357-178cef00-5ec2-11ea-8ed6-5a02e8894cb2.png)
- [ ] You **SHOULD** be redirected to `/checkout/cart` which meets criteria from below code and `restoreQuoteWhenReturningFromMollie` will be triggered
https://github.com/mollie/Magento/blob/18a6b259b186404ae0158caa65e8e46b843853a0/app/code/community/Mollie/Mpm/etc/config.xml#L130-L147
- [ ] Place another order - On checkout you will be directed to empty shopping basket page, rather than the checkout success page
- [ ] Another invoice is created for the previous order, and new order is not created.


## Notes
On `\Mage_Checkout_OnepageController::successAction` following function is triggered `\Mage_Checkout_Model_Session::clear` which clears `quote_id` and `last_success_quote_id` but `last_quote_id` which is used in `\Mollie_Mpm_Model_Observer::restoreQuoteWhenReturningFromMollie` is not cleared.

**Comment by Frank-Magmodules** [comment]
Hello @Ig0r-M,

Unfortunately, I can't reproduce the bug you described. I follow all steps, but the last 2 points you describe don't happen in our test environment:
- `On checkout you will be directed to empty shopping basket page, rather than the checkout success page`
- `Another invoice is created for the previous order, and the new order is not created.`

Do you maybe have plugins in place that hook in on the same flow? Maybe that there is a conflict with these, but on a clean Magento 1.9.4.1 installation this behavior is not reproducible.

Please fill in our [contact form](https://www.magmodules.eu/support-form.html) if your able to share your server/magento details so our team can take a look with you.

**Comment by Ig0r-M** [comment]
Thanks @Frank-Magmodules i will attempt to retest as well on an fresh Magento installation as it might be conflicting either third party code or our custom one. Unfortunately i cannot give access to the project where this has been happening.

**Comment by Frank-Magmodules** [comment]
Great @Ig0r-M , please let me know the outcome.

**Comment by AkashP454** [comment]
Apologies @Frank-Magmodules. I am a colleague of @Ig0r-M.

The last 2 steps we have pointed out seem to another issue..... (I don't think are related at the moment). The last 2 steps only occur when sometimes using another payment method (pay360)

```
- On checkout you will be directed to empty shopping basket page, rather than the checkout success page
- Another invoice is created for the previous order, and the new order is not created.
```

@Frank-Magmodules 
Were you able to replicate the fact that the quote `is_active` was being set back to `true` after an order being placed using check/mo payment (the first 5 steps written by Igor above). 

`\Mollie_Mpm_Model_Observer::restoreQuoteWhenReturningFromMollie`
 I believe this a bug as a quote shouldn't be set back to true after an order has been placed

Thanks and apologies for the confusion.

**Comment by Frank-Magmodules** [comment]
Hello @AkashP454,

Good to hear that the 2 steps are not related to our extension. About setting the quote active: Yes we have seen that and identified that it should not happen. We will look into it.

**Comment by Marvin-Magmodules** [comment]
Hi @Ig0r-M, @AkashP454, thanks again for opening this issue. The issue has been resolved in the [v5.5.6](https://github.com/mollie/Magento/releases/tag/v5.5.6) release. The quote is only restored on the right state and only for Mollie orders. We are closing this issue now but please feel free to reopen the issue if this still occurs.

**Comment by seansan** [comment]
> Hi @Ig0r-M , Thank you for opening this issue. We can't directly reproduce this. Can you contact us through our [contact form](https://www.magmodules.eu/support-form.html) so we can exchange log files to check the issue?
> Thank you!

This is the important part: a logic error and does not meet reproducing 

Plus observed is not checking if used method is actually Mollie and does this for all quotes
